guide/index.adoc

```diff
@@ -0,0 +1,929 @@
+Smaller images result in less network traffic.
+
+==== Why not just increase the hardware resources as much as possible?  I want it to run fast.
+It is critical to strike a balance between the minimum resources required to run the algorithm and acceptable runtimes. 
+Scale depends on accurate requirements in order to efficiently schedule jobs on the cluster. 
+If a job requests more resources than it uses, these resources are wasted as they cannot not be allocated to other jobs. 
+In a cloud environment, wasted resources equate to wasted money. 
+When job requirements depend on the input, the Seed manifest supports specifying a multiplicative value for the resource requests via the `inputMultiplier` property.
+
+==== My algorithm requires some reference data, should I just embed it into the Docker image?
+It depends. If the total size of the file(s) is sufficiently small; e.g. less than 100 MB, then add it into the Docker image. 
+If it has larger requirements, such as a lookup directory, please work with the Scale team to create a mountable directory.
+
+==== How to secure sensitive information like passwords inside my Docker image?
+The Seed manifest allows the developer to define "Settings", which will be defined as environment variables at runtime. 
+These "Settings" can be marked as "secret" and will be stored in {vault-url}[HashiCorp Vault]. 
+The values for these environment variables are hidden from the API and are only injected at runtime. See <<Secret Data>> for more details.
+
+==== Why should I define error codes?
+The developer should properly capture and define errors in their code using exit codes.  
+The Seed manifest allows the developer to define specific error codes with name, description, and type (job vs. data). 
+A data error indicates that the input data was invalid and the job **will never** be able to run it. 
+All other errors are job errors and could potentially be resolved on future executions (re-queues/re-process). 
+A third type, "System" errors, are reserved for the Scale framework.
+
+==== How do I use exit codes in Python/Matlab/Java/etc.?
+Code should use proper try/catch or if/else statements to identify the type of error. 
+The code can exit with any value between 0-255. 
+**The zero value indicates successful run.** 
+The exit code can be checked after a program finishes with "echo $?".
+
+Common examples exiting with code 42:
+
+Python: `sys.exit(42)`
+
+Matlab: `exit(42)`
+
+Java: `System.exit(42)`
+
+C/C++: `exit(42)`
+
+Shell: `exit 42`
+
+==== My algorithm involves running Python code, followed by Matlab code, and finally C code. Should I put it all in one Docker image?
+It depends. 
+Does each step involve heavy processing or create time-consuming intermediate products? 
+If so, they would be better as their own standalone jobs with its output products being passed to a subsequent step in a recipe. 
```

Comment:
Should we use the term workflow instead of recipe since recipe is more Scale related?

Reply:
Agree. Any time we are not specifically referencing terms with Scale, I'd prefer the more commonly recognized workflow.

Reply:
Updated in [df17722](https://github.com/ngageoint/seed-cli/pull/233/commits/df1772214e5dfcaf9c9bf7d76daa0a62fda86b0c)